assets/js/components/post-searcher.js

```diff
@@ -112,28 +113,36 @@ class PostSearcher extends Component {
 	}
 
 	onConfirm( selection ) {
-		this.setState( {
-			selection,
-		} );
+		const { results } = this.state;
+		// Check that the selection is "valid".
+		if ( Array.isArray( results ) && selection !== this.noResultsMessage ) {
+			const match = results.find( ( result ) => result.post_title === selection );
+			if ( selection && match ) {
+				this.setState( {
+					selection,
+					canSubmit: true,
+					match,
+				} );
+			}
+		} else {
+			this.setState( {
+				canSubmit: false,
+			} );
+		}
 	}
 
 	onClick() {
-		const { results, selection } = this.state;
-		const match = find(
-			results,
-			( result ) => {
-				return result.post_title === selection;
-			}
-		);
-
-		document.location = getSiteKitAdminURL(
-			'googlesitekit-dashboard',
-			{
-				id: match.id,
-				permaLink: match.permalink,
-				pageTitle: selection,
-			}
-		);
+		const { selection, match } = this.state;
+		if ( match && match.ID ) {
```

Comment:
Just wanted to note here that I needed to change this from `id` to `ID`. It looks like in the released plugin the id is not being added to the URL.

Reply:
Good catch! You're right, it's returning full `WP_Post`s, not the normal post object you would expect from a core REST controller. It looks like the ID isn't currently used, but maybe I'm missing it. Either way 👍